internal/context/context_test.go

```diff
@@ -53,6 +53,44 @@ func TestCreateContextWithFutureDeadline(t *testing.T) {
 	}
 }
 
+func TestCreateContextWithBaseContext(t *testing.T) {
+	tests := []struct {
+		deadline time.Time
```

Comment:
I think the `baseCtx` should be a field in this struct, since we may want to test with different baseCtx if we add more test cases.